aviary/subsystems/energy/test/test_battery.py

```diff
@@ -0,0 +1,81 @@
+import unittest
+
+import numpy as np
+import openmdao.api as om
+
+from openmdao.utils.assert_utils import assert_check_partials, assert_near_equal
+
+from aviary.subsystems.energy.battery_builder import BatteryBuilder
+import aviary.api as av
+
+
+class TestBatteryDerivs(unittest.TestCase):
+    def setUp(self):
+        self.prob = prob = om.Problem()
+
+        self.options = av.AviaryValues()
+
+        self.battery = BatteryBuilder()
+
+    def test_battery_premission(self):
+        prob = self.prob
+        prob.model.add_subsystem('battery_premission',
+                                 self.battery.build_pre_mission(self.options),
+                                 promotes=['*'])
+
+        prob.setup(force_alloc_complex=True)
+
+        prob.set_val(av.Aircraft.Battery.PACK_ENERGY_DENSITY, 550, units='kW*h/kg')
+        prob.set_val(av.Aircraft.Battery.PACK_MASS, 1200, units='lbm')
+        prob.set_val(av.Aircraft.Battery.ADDITIONAL_MASS, 115, units='lbm')
+
+        prob.run_model()
+
+        mass_expected = 1_315
+        energy_expected = 1_077_735_471.12
```

Comment:
maybe do this in MJ so that the number isn't quite so big

Reply:
Done!